Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1072 @@
+                var end = latestTime.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+
+                var scrip=_symbolMapper.samcoTradableSymbolList.Where(x => x.Name.ToUpperInvariant() == symbol).First();
+                string endpoint = $"/intraday/candleData?symbolName={symbol}&fromDate={start}&toDate={end}";
+                if (scrip.Instrument == "INDEX")
+                {
+                    endpoint= $"/intraday/indexCandleData?symbolName={symbol}&fromDate={start}&toDate={end}";
+                }
+                var restRequest = new RestRequest(endpoint, Method.GET);
+                var response = _samcoAPI.ExecuteRestRequest(restRequest);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception(
+                        $"SamcoBrokerage.GetHistory: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, " +
+                        $"Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+                }
+
+                // we need to drop the last bar provided by the exchange as its open time is a history request's end time
+                var candles = JsonConvert.DeserializeObject<CandleResponse>(response.Content);
+
+
+                if (candles.intradayCandleData?.Any() == null)
+                {
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "NoHistoricalData",
+                        $"Exchange returned no data for {symbol} on history request " +
+                        $"from {request.StartTimeUtc:s} to {request.EndTimeUtc:s}"));
+                    yield break;
+                }
+
+                foreach (var candle in candles.intradayCandleData)
+                {
+                    if (!exchange.DateTimeIsOpen(candle.dateTime) && !request.IncludeExtendedMarketHours)
+                    {
+                        continue;
+                    }
+                    yield return new TradeBar()
+                    {
+                        Time = candle.dateTime,
+                        Symbol = request.Symbol,
+                        Low = candle.low,
+                        High = candle.high,
+                        Open = candle.open,
+                        Close = candle.close,
+                        Volume = candle.volume,
+                        Value = candle.close,
+                        DataType = MarketDataType.TradeBar,
+                        Period = period,
+                        EndTime = candle.dateTime.AddMinutes(1)
+                    };
+                }
+            } while (latestTime < request.EndTimeUtc);
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public override void Dispose()
+        {
+        }
```

Comment:
At least have to dispose of the `_samcoAPI`, my guess is that there are probably some other component to dispose of, let's review it 👍